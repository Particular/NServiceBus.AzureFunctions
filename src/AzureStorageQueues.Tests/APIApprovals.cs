namespace AzureStorageQueues.Tests
{
    using NServiceBus.AzureFunctions.AzureStorageQueues;
    using NUnit.Framework;
    using Particular.Approvals;
    using PublicApiGenerator;

    [TestFixture]
    public class APIApprovals
    {
        [Test]
        public void Approve()
        {
            var publicApi = ApiGenerator.GeneratePublicApi(typeof(AzureStorageQueueTriggerEndpoint).Assembly, excludeAttributes: new[] { "System.Runtime.Versioning.TargetFrameworkAttribute" });
            Approver.Verify(publicApi);
        }
    }
}