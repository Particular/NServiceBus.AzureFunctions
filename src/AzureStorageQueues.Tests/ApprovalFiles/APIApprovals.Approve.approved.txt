namespace NServiceBus.AzureFunctions.AzureStorageQueues
{
    public class AzureFunctionEndpoint : NServiceBus.Serverless.ServerlessEndpoint<Microsoft.Azure.WebJobs.ExecutionContext>
    {
        public AzureFunctionEndpoint(System.Func<Microsoft.Azure.WebJobs.ExecutionContext, NServiceBus.Serverless.ServerlessEndpointConfiguration> configurationFactory) { }
    }
    public class AzureStorageQueueTriggeredEndpointConfiguration : NServiceBus.Serverless.ServerlessEndpointConfiguration
    {
        public AzureStorageQueueTriggeredEndpointConfiguration(string endpointName) { }
    }
    public class static AzureStorageQueueTriggerExtensions
    {
        public static System.Threading.Tasks.Task Process(this NServiceBus.AzureFunctions.AzureStorageQueues.AzureFunctionEndpoint endpoint, Microsoft.WindowsAzure.Storage.Queue.CloudQueueMessage message, Microsoft.Azure.WebJobs.ExecutionContext executionContext) { }
    }
}